### ISSUE TITLE
Broken ABS integration?

### ISSUE THREAD

**Issue by Artnal** [IgnisDa/ryot#1049]
Hi there, 

I have been wondering if something had broken a few weeks ago in the audiobookshelf integration. For some reason, even though it tells me that it keeps regularly running, no updates to the audiobooks I'm listening to seem to appear in Ryot. 

At first, I thought the issue could have been a specific book, but after the fourth book in a row that does have an audible ASIN, yet does not report to Ryot, I'm starting to wonder if the issue lies somewhere else. I've also tried recreating the integration, to no avail. 

Is everything still working on your end?

Cheers

**Comment by IgnisDa** [comment]
It seems to be working on my end. If possible, can you share credentials to your ABS instance so that I can reproduce the issue?

**Comment by IgnisDa** [comment]
before you do that, pls try setting env variable `RUST_LOG=ryot=debug` and try triggering an integration sync from the miscellaneous settings.

**Comment by IgnisDa** [comment]
The progress percentage says 42. Is that correct for this audiobook?

**Comment by IgnisDa** [comment]
I think that is the log of the data that is returned by ABS, so it might be possible that your ABS instance is misbehaving. Can you take a look? Maybe update it to the latest version.

Also please force a sync from the miscellaneous settings and then attach the container logs.

**Comment by IgnisDa** [comment]
I meant the ryot container logs assuming you have the rust log env variable set to debug.

**Comment by IgnisDa** [comment]
What is the expected behaviour and what is happenning right now?

**Comment by IgnisDa** [comment]
I will be releasing a new version soon that contains more logging. Meanwhile can you update to version `v7.4` which contains a few changes to the caching mechanism?

**Comment by IgnisDa** [comment]
Please upgrade to v7.5.0 when it is released this weekend and paste the logs here. It has logging about when a progress update is rejected.

**Comment by IgnisDa** [comment]
@Artnal Released 7.5.0. Please paste the logs here when you trigger a sync from the miscellaneous settings.

**Comment by IgnisDa** [comment]
> No progress update required

This happens when the progress is already updated (https://github.com/IgnisDa/ryot/blob/8a2297a500a4bb77f79872d2cba8dbcd7d851967/crates/services/miscellaneous/src/lib.rs#L1203).

Are you sure the updates are not working? Can you attach a screenshot of history tab of `https://ryot-your-instance.com/media/item/met_Db5Q-6ymfyuQ`?

**Comment by IgnisDa** [comment]
What is the progress of these 2 audiobooks in ABS?

**Comment by IgnisDa** [comment]
Can you share your docker compose? Make sure you redact secrets/

**Comment by IgnisDa** [comment]
Also run this SQL: `select * from integration`. Make sure you redact the credentials.

**Comment by IgnisDa** [comment]
You can run the SQL using `docker exec -u postgres -it ryot-db psql` and then pasting the sql i sent above.

In your docker compose, you can get rid of `FRONTEND_INSECURE_COOKIES`. It is no longer used.

**Comment by IgnisDa** [comment]
Please add a semicolon at the end

**Comment by IgnisDa** [comment]
Ah so this is a configuration mistake. As you can see, you have set the `minimum_progress` to 1 and `maximum_progress` to 90. Anything outside these ranges will not be synced. Please edit your integration for your usecase.

This is done because some movies, audiobooks etc end before 100% and users don't always complete 100%.

Closing since everything is working as expected.

**Comment by IgnisDa** [comment]
You are right. Can you join the discord? I will need to debug this on my end.

**Comment by IgnisDa** [comment]
Closing since this is proving impossible to debug. Feel free to ping me here with updates.

**Comment by IgnisDa** [comment]
~~god~~ software works in mysterious ways
